frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx

```diff
@@ -0,0 +1,188 @@
+      }
+    };
+  };
+
+  return (
+    <Form>
+      <FormGroup 
+        label={t('ceph-storage-plugin~Endpoint')} 
+        fieldId="endpoint-input"
+        isRequired
+        validated = {endpointValid}
+        helperText={
+          t('ceph-storage-plugin~Rest API IP address of IBM Storage FlashSystem.')
+        }
+        helperTextInvalid={t('ceph-storage-plugin~The endpoint is not a valid URL')}
+        >
+        <TextInput
+          id="endpoint-input"
+          value={formState.endpoint}
+          type="text"
+          onChange={onChange}
+          isRequired
+        />
+      </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Username')} isRequired fieldId="username-input">
+        <TextInput
+          id="username-input"
+          value={formState.username}
+          type="text"
+          onChange={(value: string) => setFormState('username', value)}
+          isRequired
+        />
+      </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Password')} isRequired fieldId="password-input" >
+      {reveal ? (
+        <>
+        <TextInput
+          id="password-input"
+          value={formState.password}
+          type="text"
+          onChange={(value: string) => setFormState('password', value)}
+          isRequired
+        />
+                
+        </>
+        ) : (
+        <>
+        <TextInput
+          id="password-input"
+          value={formState.password}
+          type="password"
+          onChange={(value: string) => setFormState('password', value)}
+          isRequired
+        />
+        </>
+        )}
+        <Button
+            type="button"
+            onClick={() => setReveal(!reveal)}
+            variant="link"
+            className="pf-m-link--align-right"
+          >
+            {reveal ? (
+              <>
+                <EyeSlashIcon className="co-icon-space-r" />
+                {t('ceph-storage-plugin~Hide Values')}
+              </>
+            ) : (
+              <>
+                <EyeIcon className="co-icon-space-r" />
+                {t('ceph-storage-plugin~Reveal Values')}
+              </>
+            )}
+          </Button>
+      </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Poolname')} isRequired fieldId="poolname-input">
+        <TextInput
+          id="poolname-input"
+          value={formState.poolname}
+          type="text"
+          onChange={(value: string) => setFormState('poolname', value)}
+          isRequired
+        />
+      </FormGroup>
+    </Form>
+  );
+};
+
+export const FlashsystemPayload: CreatePayload<FlashsystemState> = (systemName, form, model, storageClassName) => {
+  const namespace = 'openshift-storage';
+  const defaultFilesystem = 'ext4';
+  const defaultVolumeMode = 'thick';
+  const defaultVolumePrefix = 'odf';
+
+  const IBMFlashsystemTemplate: IBMFlashsystemKind = {
+    apiVersion: apiVersionForModel(IBMFlashsystemModel),
+    kind: IBMFlashsystemModel.kind,
```

Comment:
apiGroup would also be needed IMO
```suggestion
    apiVersion: apiVersionForModel(IBMFlashsystemModel),
    apiGroup: 
    kind: IBMFlashsystemModel.kind,
```